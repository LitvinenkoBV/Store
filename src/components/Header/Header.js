import "./Header.scss";
import FavoritesHeader from "../FavoritesHeader/FavoritesHeader";
import Basket from "../Basket/Basket";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Header = () => {
  return (
    <>
      <header className="header">
        <Link to="/">
          <div className="header__logo">
            <img src="./img/logo.svg" alt="" />
          </div>
        </Link>

        <div>
          <Basket />
          <FavoritesHeader />
        </div>
      </header>
    </>
  );
};

Header.propTypes = {
  arrProductsToBuy: PropTypes.array,
};

export default Header;
