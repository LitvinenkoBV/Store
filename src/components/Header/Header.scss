.header {
  background-color: #b99b9b;
  padding: 20px;
  margin-bottom: 50px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}
