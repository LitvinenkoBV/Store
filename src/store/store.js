import { createStore, combineReducers, compose, applyMiddleware } from "redux";
import thunk from "redux-thunk";
import { productsReducer } from "./products/reducer";
import { getFavoriteProductsReducer } from "./favorite_products//reducer";
import { productsToBuyReducer } from "./products_to_buy/reducer";
import { modalReducer } from "./modal/reducer";

export const rootReducer = combineReducers({
  products: productsReducer,
  favoriteProducts: getFavoriteProductsReducer,
  productsToBuy: productsToBuyReducer,
  modal: modalReducer,
});

const devTools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f;

export const store = createStore(
  rootReducer,
  compose(applyMiddleware(thunk), devTools)
);
