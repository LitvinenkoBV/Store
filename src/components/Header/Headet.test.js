import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { legacy_createStore as createStore } from "redux";
import Header from "./Header";
import { rootReducer } from "../../store/store";
import { BrowserRouter } from "react-router-dom";

const store = createStore(rootReducer, {});

describe("<Header/> component", () => {
  const { container } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );
  it("should render", () => {
    expect(container.firstChild).toMatchSnapshot();
  });
});
