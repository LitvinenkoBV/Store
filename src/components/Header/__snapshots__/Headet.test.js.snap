// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> component should render 1`] = `
<header
  class="header"
>
  <a
    href="/"
  >
    <div
      class="header__logo"
    >
      <img
        alt=""
        src="./img/logo.svg"
      />
    </div>
  </a>
  <div>
    <a
      class="header__basket"
      href="/basket"
    >
      <img
        alt=""
        src="./img/shopping-cart.png"
      />
      0
    </a>
    <a
      class="header__favorites"
      href="/favorites"
    >
      <img
        alt=""
        src="./img/favorite.png"
      />
      0
    </a>
  </div>
</header>
`;
